durable-storage/src/database.rs

```diff
@@ -118,3 +121,53 @@ impl Database {
         value.ok_or(DatabaseError::KeyNotFound)
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use std::sync::Arc;
+
+    use bytes::Bytes;
+    use tokio::runtime::Handle;
+
+    use super::Database;
+    use crate::merkle_layer::Key;
+    use crate::merkle_worker::MerkleWorker;
+    use crate::persistence_layer::PersistenceLayer;
+    use crate::persistence_layer::utils::TestableTmpdir;
+    use crate::repo::DirectoryManager;
+
+    fn new_database(handle: &Handle) -> Database {
+        let tmpdir = TestableTmpdir::new();
+
+        let repo =
+            DirectoryManager::new(tmpdir.path()).expect("Failed to create directory manager");
+
+        let persistent: Arc<PersistenceLayer> = PersistenceLayer::new(&repo)
+            .expect("Creating a persistence layer should succeed")
+            .into();
+
+        let merkle = MerkleWorker::new(handle, persistent.clone())
+            .expect("Creating a Merkle worker should succeed");
+
+        Database { persistent, merkle }
+    }
+
+    #[test]
+    fn test_database_write_zero_offset() {
```

Comment:
Maybe worth having this as a property-based test with various keys and data to write?

Reply:
You could vary the length of the key and data too with something like 

```rust
        fn test_database_write_zero_offset(
            keys in prop::collection::vec(
                prop::collection::vec(any::<u8>(), 0..=KEY_MAX_SIZE),
                0..100),
```